packages/next-swc/crates/next-custom-transforms/src/transforms/react_server_components.rs

```diff
@@ -540,7 +540,11 @@ impl ReactServerComponentValidator {
                 JsWord::from("react-dom/server"),
                 JsWord::from("next/router"),
             ],
-            invalid_client_imports: vec![JsWord::from("server-only"), JsWord::from("next/headers")],
+            invalid_client_imports: vec![
+                JsWord::from("server-only"),
+                JsWord::from("next/headers"),
+                JsWord::from("next/server"),
```

Comment:
the goal here is to ban usage in pages and client modules, but i'm not sure if this is the right way to do it -- might disallow valid uses of `next/server` in too many places.

Reply:
I feel we should create a new one for invalid client apis for `next/server`, as there're some API is fine to use in pages like `userAgent`, `userAgentFromString` and `URLPattern`.

Like create a new `invalid_client_lib_apis_mapping`. I feel fine if we need to ship this separately and just throw a runtime error for `after()` usage in pages router.

Reply:
Added `invalid_client_lib_apis_mapping`, pretty much a straight copy from `invalid_server_lib_apis_mapping`. lmk if that works.